tests/dotnet/UnitTests/ProjectTest.cs

```diff
@@ -682,6 +684,173 @@ public void AppWithResources (ApplePlatform platform, string runtimeIdentifiers)
+
+			var assetsCar = Path.Combine (resourcesDirectory, "Assets.car");
+			Assert.That (assetsCar, Does.Exist, "Assets.car");
+
+			var doc = ProcessAssets (assetsCar);
+			Assert.IsNotNull (doc, "There was an issue processing the asset binary.");
+			var foundAssets = FindAssets (doc);
+
+			var TotalUniqueAssets = 18;
+			Assert.AreEqual (TotalUniqueAssets, foundAssets.Count, "There were a different number of assets found.");
+			Assert.IsFalse (foundAssets.Contains ("Data.DS_StoreDataTest"), "DS_Store files should not be included.");
+
+			var arm64txt = Path.Combine (resourcesDirectory, "arm64.txt");
+			var armtxt = Path.Combine (resourcesDirectory, "arm.txt");
+			var x64txt = Path.Combine (resourcesDirectory, "x64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm64")), File.Exists (arm64txt), "arm64.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-arm")), File.Exists (armtxt), "arm.txt");
+			Assert.AreEqual (runtimeIdentifiers.Split (';').Any (v => v.EndsWith ("-x64")), File.Exists (x64txt), "x64.txt");
+		}
+
+		void DeleteAssets (string project_path)
+		{
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+			try {
+				if (Directory.Exists (xcassetsDir))
+					Directory.Delete (xcassetsDir, true);
+			}
+			catch (Exception e){
+				Console.WriteLine($"Deleting Assets.xcassets failed: {e.Message}");
+			}
+		}
+
+		void CopyAssets (string project_path)
+		{
+			DirectoryInfo testingAssetsDir = new DirectoryInfo (Path.Combine (project_path, @"../../TestingAssets"));
+			string xcassetsDir = Path.Combine (project_path, @"../Assets.xcassets");
+
+			if (!testingAssetsDir.Exists)
+				throw new DirectoryNotFoundException ($"TestingAssets directory does not exist or could not be found: {testingAssetsDir}");
+
+			if (!Directory.Exists (xcassetsDir))
+				CopyDirectoryContents (testingAssetsDir, xcassetsDir);
+		}
+
+		void CopyDirectoryContents (DirectoryInfo dir, string DestDirectory) {
+			Directory.CreateDirectory (DestDirectory);
+			foreach (FileInfo file in dir.GetFiles ()) {
+				string tempPath = Path.Combine (DestDirectory, file.Name);
+				file.CopyTo (tempPath, false);
+			}
+			foreach (DirectoryInfo subDir in dir.GetDirectories ()){
+				CopyDirectoryContents (subDir, Path.Combine (DestDirectory, subDir.Name));
+			}
+		}
+
+		JsonDocument ProcessAssets (string assetsPath) {
+			try {
+				using (Process myProcess = new Process ()) {
+					myProcess.StartInfo.UseShellExecute = false;
+					myProcess.StartInfo.RedirectStandardOutput = true;
+					myProcess.StartInfo.Arguments = $"--sdk iphoneos assetutil --info {assetsPath}";
+					myProcess.StartInfo.FileName = "xcrun";
+					myProcess.StartInfo.CreateNoWindow = true;
+					myProcess.Start ();
+
+					string output = myProcess.StandardOutput.ReadToEnd ();
+					StreamReader reader = myProcess.StandardOutput;
+					myProcess.WaitForExit ();
```

Comment:
This can block the test FOR EVER if the process that is being executed does not exit, you need to use https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.waitforexit?view=net-5.0#System_Diagnostics_Process_WaitForExit_System_Int32_ and pass a timeout. Then check the result if it is true of false and assert for it to be false.

You can do that in several ways, use an out parameter, to state if the process exited correctly or move the return type to JsonDocument? and assert is not null.